x-pack/plugins/ingest_manager/server/services/es_index_pattern.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { SavedObjectsClientContract } from 'kibana/server';
+import { getInstallation } from './epm/packages/get';
+
+export interface ESIndexPatternService {
+  getESIndexPattern(
+    savedObjectsClient: SavedObjectsClientContract,
+    pkgName: string,
+    datasetPath: string
+  ): Promise<string | undefined>;
+}
+
+export class ESIndexPatternSavedObjectService implements ESIndexPatternService {
```

Comment:
side note, classes in ts are also interfaces, so `ESIndexPatternSavedObjectService` can be used exactly the same as `ESIndexPatternService`https://www.typescriptlang.org/docs/handbook/classes.html#using-a-class-as-an-interface

Reply:
Ah cool! Didn't realize that.